bokeh/models/widgets/widget.py

```diff
@@ -17,3 +17,15 @@ class Widget(LayoutDOM):
     ''' A base class for all interactive widget types.
 
     '''
+
+class NeedsJQuery(object):
+    """ A mixin for widgets requiring JQuery and/or JQuery UI. """
+
+    __javascript__ = [
+        "https://code.jquery.com/jquery-3.2.1.min.js",
+        "https://code.jquery.com/ui/1.12.1/jquery-ui.min.js",
+    ]
+
+    __css__ = [
+        "https://code.jquery.com/ui/1.12.1/themes/base/jquery-ui.css",
+    ]
```

Comment:
At this point I think that the only viable option is to drop jquery-ui completely. I already reimplemented autocomplete (naively, but can be improved later). I don't care about date picker, so it can be either removed or rewritten (if someone want to devote time to this). This leaves us with SlickGrid. I see that it uses jquery-ui only for sortable column and some trivia (icons), so I should be able to fix this in a day or two.